src/Storages/ExternalTable/StorageExternalTable.cpp

```diff
@@ -0,0 +1,99 @@
+#include <Common/logger_useful.h>
+#include <Interpreters/Context.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <Storages/ExternalTable/ClickHouse/ClickHouse.h>
+#include <Storages/ExternalTable/ExternalTableFactory.h>
+#include <Storages/ExternalTable/StorageExternalTable.h>
+
+namespace DB
+{
+
+StorageExternalTable::StorageExternalTable(
+        std::unique_ptr<ExternalTableSettings>  settings,
+        const StorageFactory::Arguments & args)
+: IStorage(args.table_id)
+, WithContext(args.getContext()->getGlobalContext())
+{
+    external_table = ExternalTableFactory::instance().getExternalTable(args.table_id.getTableName(), std::move(settings));
+
+    /// First, setStorageMetadata should be allowed to fail (the only failable part is getTableStructure function call), otherwise it will block Proton from starting up.
+    /// Second, when it fails, the exception should be caught, otherwise, Proton will fail to start.
+    /// TODO we could use cache to save the table structure, so that when Proton restarts it could read from the cache directly.
+    try
+    {
+        setStorageMetadata(args);
+    }
+    catch (const Exception & e)
+    {
+        LOG_ERROR(&Poco::Logger::get("ExternalTable-ClickHouse" + args.table_id.getFullTableName()),
```

Comment:
Probably we can just use `ExternalTable` as the logger name and log the details in the message